src/Lasso.jl

```diff
@@ -301,10 +322,191 @@ function StatsBase.fit{T<:AbstractFloat,V<:FPVector}(::Type{LassoPath},
+    Note that for GammaLassoPath this may be a crude approximation, as gamlr does this differently.
+"""
+function df(path::RegularizationPath)
+    nλ = length(path.λ)
+    βs = coef(path)
+    dof = zeros(Int,nλ)
+    for s=1:nλ
+        dof[s] = sum(βs[:,s].!=0)
+    end
+
+    if dispersion_parameter(path)
+        # add one for dispersion_parameter
+        dof+=1
+    end
+
+    dof
+end
+
+function aicc(path::RegularizationPath;k=2)
+    d = df(path)
+    n = nobs(path)
+    ic = -2loglikelihood(path) + k*d + k*d.*(d+1)./(n-d-1)
+    ic[d.+1 .> n] = realmax(eltype(ic))
+    ic
+end
+
+minAICc(path::RegularizationPath;k=2)=indmin(aicc(path;k=k))
+
+hasintercept(path::RegularizationPath) = hasintercept(path.m.pp)
+
+#Consistent with StatsBase.coef, if the model has an intercept it is included.
+function StatsBase.coef(path::RegularizationPath; select=:all, nCVfolds=10)
+    if length(path.λ) == 0
+        X = path.m.pp.X
+        p = size(X,2)
+        if hasintercept(path)
+            p+=1
+        end
+        return zeros(eltype(X),p)
+    end
+
+    if select == :all
+        if hasintercept(path)
+            vcat(path.b0',path.coefs)
+        else
+            path.coefs
+        end
+    elseif select == :AICc
+        if hasintercept(path)
+            vec(vcat(path.b0[minAICc(path)],path.coefs[:,minAICc(path)]))
+        else
+            path.coefs[:,minAICc(path)]
+        end
+    elseif select == :CVmin || select == :CV1se
+        gen = Kfold(length(path.m.rr.y),nCVfolds)
+        segCV = cross_validate_path(path;gen=gen,select=select)
+        if hasintercept(path)
+            vec(vcat(path.b0[segCV],path.coefs[:,segCV]))
+        else
+            path.coefs[:,segCV]
+        end
+    else
+        error("unknown selector $select")
+    end
+end
+
+# function intercept(path::RegularizationPath; select=:all)
+#     if select == :all
+#         path.b0
+#     elseif select == :AICc
+#         path.b0[minAICc(path)]
+#     end
+# end
+
+## Prediction function for GLMs
+function StatsBase.predict{T<:AbstractFloat}(path::RegularizationPath, newX::AbstractMatrix{T}; offset::FPVector=Array(T,0), select=:all)
+    # add an interecept to newX if the model has one
+    if hasintercept(path)
+        newX = [ones(eltype(newX),size(newX,1),1) newX]
+    end
+
+    # calculate etas for each obs x segment
+    eta = newX * coef(path;select=select)
+
+    # get model
+    mm = path.m
+
+    # adjust for any offset
+    if length(mm.rr.offset) > 0
+        length(offset) == size(newX, 1) ||
+            throw(ArgumentError("fit with offset, so `offset` kw arg must be an offset of length `size(newX, 1)`"))
+        broadcast!(+, eta, eta, offset)
+    else
+        length(offset) > 0 && throw(ArgumentError("fit without offset, so value of `offset` kw arg does not make sense"))
+    end
+
+    if typeof(path.m) <: LinearModel
+        eta
+    else
+        # invert all etas to mus
+        μ(η) = linkinv(mm.rr.l, η)
+        map(μ,eta)
```

Comment:
That's a sign that there is a [type stability issue](http://docs.julialang.org/en/release-0.4/manual/performance-tips/#write-type-stable-functions) somewhere, probably with `coef` unless you already changed it. You can use `T[linkinv(m.rr.l, η) for η in eta]` which will always return an array of type T, and on Julia 0.5 you won't even need this change, but it would still be good to fix the type stability issue eventually since it will have a performance impact.